core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -917,12 +929,12 @@ public void success(TaskInfo value)
             try (SetThreadName ignored = new SetThreadName("UpdateResponseHandler-%s", taskId)) {
                 try {
                     long currentRequestStartNanos;
-                    synchronized (HttpRemoteTask.this) {
+                    synchronized (httpClient) {
```

Comment:
I do not get this. Now there are places where you modify `currentRequest` without holding `httpClient` lock.

Also with IntelliJ reports lots of:
![image](https://user-images.githubusercontent.com/890111/187647474-d9aabe1e-5360-4eeb-a351-34dfa3ac5611.png)


Please enable those inspections:
![image](https://user-images.githubusercontent.com/890111/187647980-22a9330d-42c8-4c89-acd6-95c22f531549.png)

and fix the code so warnings are not reported